pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -123,13 +134,43 @@ type FlowExporter struct {
 	exporterInput          exporter.ExporterInput
 	k8sClient              kubernetes.Interface
 	nodeRouteController    *noderoute.Controller
+	l7VisibilityController *l7engine.L7VisibilityController
 	isNetworkPolicyOnly    bool
 	nodeName               string
 	conntrackPriorityQueue *priorityqueue.ExpirePriorityQueue
 	denyPriorityQueue      *priorityqueue.ExpirePriorityQueue
 	expiredConns           []flowexporter.Connection
 	egressQuerier          querier.EgressQuerier
 	podStore               podstore.Interface
+	l7events               map[flowexporter.ConnectionKey]map[int]Http
+	l7mut                  sync.Mutex
+}
+
+// Define struct to hold the L7 flow JSON values
+type Http struct {
+	Hostname      string `json:"hostname"`
+	URL           string `json:"url"`
+	UserAgent     string `json:"http_user_agent"`
+	ContentType   string `json:"http_content_type"`
+	Method        string `json:"http_method"`
+	Protocol      string `json:"protocol"`
+	Status        int    `json:"status"`
+	ContentLength int    `json:"length"`
+}
+
+type JsonToEvent struct {
```

Comment:
I believe this format is defined by Suricata? Could you add a comment to this effect, and if there is any Suricata documentation that describes this format, include a link to it in your comment?

Reply:
Added the reference